src/value/tests.rs

```diff
@@ -227,6 +235,83 @@ fn from_c_mg_value_string() {
     assert_eq!(format!("{}", mg_value), "\'ṰⱻⱾᵀ\'");
 }
 
+#[test]
+fn from_c_mg_value_date1() {
+    let c_date = bindings::mg_date { days: 100 };
+    let c_mg_value = unsafe { bindings::mg_value_make_date(bindings::mg_date_copy(&c_date)) };
+    let mg_value = unsafe { Value::from_mg_value(c_mg_value) };
+    assert_eq!(
+        Value::Date(
+            NaiveDate::from_ymd(1970, 1, 1)
+                .checked_add_signed(Duration::days(100))
+                .unwrap()
+        ),
+        mg_value
+    );
+    assert_eq!(format!("{}", mg_value), "'1970-04-11'");
+}
+
+#[test]
+fn from_c_mg_value_date2() {
+    let c_date = bindings::mg_date { days: 365 };
+    let c_mg_value = unsafe { bindings::mg_value_make_date(bindings::mg_date_copy(&c_date)) };
+    let mg_value = unsafe { Value::from_mg_value(c_mg_value) };
+    assert_eq!(Value::Date(NaiveDate::from_ymd(1971, 1, 1)), mg_value);
+    assert_eq!(format!("{}", mg_value), "'1971-01-01'");
+}
+
+#[test]
+fn from_c_mg_value_local_time() {
+    let c_local_time = bindings::mg_local_time {
+        nanoseconds: 52835851241000,
+    };
+    let c_mg_value =
+        unsafe { bindings::mg_value_make_local_time(bindings::mg_local_time_copy(&c_local_time)) };
+    let mg_value = unsafe { Value::from_mg_value(c_mg_value) };
+    assert_eq!(
+        Value::LocalTime(NaiveTime::from_hms_micro(14, 40, 35, 851241)),
+        mg_value
+    );
+    assert_eq!(format!("{}", mg_value), "'14:40:35.851241'");
+}
+
+#[test]
+fn from_c_mg_value_local_date_time() {
+    let c_local_date_time = bindings::mg_local_date_time {
+        seconds: 500 * 24 * 60 * 60 + 52835,
+        nanoseconds: 851241 * 1000,
+    };
+    let c_mg_value = unsafe {
+        bindings::mg_value_make_local_date_time(bindings::mg_local_date_time_copy(
+            &c_local_date_time,
+        ))
+    };
+    let mg_value = unsafe { Value::from_mg_value(c_mg_value) };
+    assert_eq!(
+        Value::LocalDateTime(NaiveDate::from_ymd(1971, 5, 16).and_hms_micro(14, 40, 35, 851241)),
```

Comment:
Could you please add one more test where the date is before the unix epoch ?

Reply:
I did that, but is this a correct result/test? What's the expected behavior when `nanoseconds` are negative? In this case, `NaiveDateTime::from_timestamp_opt` throw/returns `None` because nanoseconds are in negative range.

Reply:
`nanoseconds` will never be negative (because of how we serialize `LocalDateTime`). However, `seconds can be negative`. For example: `LocalDateTime{.seconds = -1, .nanoseconds = 0}` should produce the timepoint `1969-12-31-T00:00`

Reply:
I'll add that exact test :smile: